pkg/apis/softwarecomposition/types.go

```diff
@@ -216,3 +216,109 @@ type VulnerabilityManifestSummaryList struct {
 
 	Items []VulnerabilityManifestSummary
 }
+
+// -------------- configuration scanning types -----------------
+
+// WorkloadConfigurationScan is a list of workload configuration scan results.
+type WorkloadConfigurationScanList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+
+	Items []WorkloadConfigurationScan
+}
+
+// WorkloadConfigurationScan is a custom resource that describes a configuration scan result of a workload.
+type WorkloadConfigurationScan struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec WorkloadConfigurationScanSpec
+}
+
+type WorkloadConfigurationScanSpec struct {
+	Controls       map[string]ScannedControl
+	RelatedObjects []WorkloadScanRelatedObject
+}
+
+type ScannedControl struct {
+	ControlID string
+	Name      string
+	Severity  ControlSeverity
+	Status    ScannedControlStatus
+	Rules     []ScannedControlRule
+}
+
+type ControlSeverity struct {
+	Severity    string
+	ScoreFactor float32
+}
+
+type ScannedControlStatus struct {
+	Status    string
+	SubStatus string
+	Info      string
+}
+
+type ScannedControlRule struct {
+	Name                  string
+	Status                RuleStatus
+	ControlConfigurations map[string][]string
+	Paths                 []RulePath
+	AppliedIgnoreRules    []string
+	RelatedResourcesIDs   []string // ?
```

Comment:
@dwertent not sure about this field -  https://github.com/kubescape/opa-utils/blob/master/reporthandling/results/v1/resourcesresults/datastructures.go#L35